example/src/cat/cats.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from 'nestjs-typegoose';
+import { Cat } from './cat.model';
+import { ReturnModelType } from '@typegoose/typegoose';
+
+@Injectable()
+export class CatsService {
+  constructor(
+    @InjectModel(Cat) private readonly catModel: ReturnModelType<typeof Cat>,
+  ) {}
+
+  async create(createCatDto: { name: string }): Promise<Cat> {
+    const createdCat = new this.catModel(createCatDto);
```

Comment:
```suggestion
    return await this.catModel.create(createCatDto);
```

Reply:
The docs should probably be updated as well.